### ISSUE TITLE
Possible bug after SetComponent causing panic: runtime error: runtime error: index out of range [1] with length 1

### ISSUE THREAD

**Issue by AlexeyDsov** [yohamta/donburi#14]
Hello!

I'm using the version github.com/yohamta/donburi v1.1.1

I'm not sure if this is a bug, maybe I'm using the library incorrectly. I have the following scenario - I create an entity, then add another component to it.

Then I repeat these steps again and at the stage of adding the second component I already get a panic: runtime error: index out of range [1] with length 1.

Example of reproducing an error:

https://gist.github.com/AlexeyDsov/78e2f499d96700f163bf87915cfa2b54

Stack trace below:
```
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/yohamta/donburi/internal/storage.(*Archetype).SwapRemove(...)
        .../go/pkg/mod/github.com/yohamta/donburi@v1.1.1/internal/storage/archetype.go:39
github.com/yohamta/donburi.(*world).TransferArchetype(0xc000110000?, 0xc000010280?, 0x2?, 0x2?)
        .../go/pkg/mod/github.com/yohamta/donburi@v1.1.1/world.go:176 +0xb65
github.com/yohamta/donburi.(*Entry).AddComponent(0xc0000660e0, 0xc00000c048, {0xc00007af68, 0x1, 0x52ec60?})
        .../go/pkg/mod/github.com/yohamta/donburi@v1.1.1/entry.go:80 +0x166
github.com/yohamta/donburi.Add[...](...)
        .../go/pkg/mod/github.com/yohamta/donburi@v1.1.1/entry.go:26
main.main()
        .../go/src/alexeydsov/test_world/main.go:28 +0x3d5

```

**Comment by yohamta** [comment]
Thank you for reporting the issue! I will investigate it tomorrow.

**Comment by yohamta** [comment]
@AlexeyDsov Hi, the bug is now fixed in v1.1.2.
I also added more tests to catch bugs. Thanks for your patience! :)
